pkg/store/labelpb/label.go

```diff
@@ -28,34 +25,36 @@ var (
 	sep = []byte{'\xff'}
 )
 
-func noAllocString(buf []byte) string {
-	return *(*string)(unsafe.Pointer(&buf))
-}
-
 func noAllocBytes(buf string) []byte {
 	return *(*[]byte)(unsafe.Pointer(&buf))
 }
 
-// ZLabelsFromPromLabels converts Prometheus labels to slice of labelpb.ZLabel in type unsafe manner.
-// It reuses the same memory. Caller should abort using passed labels.Labels.
-func ZLabelsFromPromLabels(lset labels.Labels) []ZLabel {
-	return *(*[]ZLabel)(unsafe.Pointer(&lset))
+// ProtobufLabelsFromPromLabels converts Prometheus labels to a slice pointers to labelpb.Label.
+func ProtobufLabelsFromPromLabels(lset labels.Labels) []*Label {
```

Comment:
This kinda defeats the purpose of ZLabels. Can we use the protobuf definitions from Prometheus directly?

Reply:
I don't think we can since Prometheus protobuf is generated with gogoprotobuf and that interferes with our code. Worth testing out but I think we might get a similar panic to this [one](https://github.com/gogo/protobuf/issues/744).